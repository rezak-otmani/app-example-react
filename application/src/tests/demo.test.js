import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Demo from '../views/demo/demo';
import { Provider } from "react-redux";
import store from '../redux/store.js';

describe('Demo Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Demo />
    );
  });

           it('it should render Demo components ', () => { 
              expect(wrapper.length).toBe(1);

          });
          
           it('it should render the title', () => {
             const item = wrapper.find('.content').at(0);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('REQUEST DEMO')
           });

        
describe('Form section', () => {

          it('it should render the form demo', () => {
               const form = wrapper.find('Form').first();
               expect((form).length).toBe(1);
 
          });

});

describe('Logo section', () => {

          it('it should render the logo ', () => {

              expect(wrapper.find('svg').length).toBe(1);
           });


});

describe('LINKS section', () => {
           
            it('it shows a list of links ', () => {
             const link = wrapper.find('.content').at(1);
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('LINKS')
           });  
      

            it('shows a link to Home', () => {
             const link = wrapper.find({to: ''});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Home ')
           });
 
            it('shows a link to Product', () => {
             const link = wrapper.find({to: 'product'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Product')
           });

           it('shows a link to About', () => {
             const link = wrapper.find({to: 'about'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('About')
           }); 

             it('shows a link to Featured', () => {
             const link = wrapper.find({to: 'featured'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Featured')
           });

                it('shows a link to Contact', () => {
             const link = wrapper.find({to: 'contact'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Contact')
           });

              it('shows a link to Demo', () => {
             const link = wrapper.find({to: 'demo'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Demo')
           });


});


describe('ABOUT section', () => {

             it('it shows about section ', () => {
             const item = wrapper.find('.content').at(2);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('ABOUT')
              });

           it('shows item mail ', () => {
             const item = wrapper.find('.item').at(6);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('info@mysite.com ')
           });
           
            it('shows item tel', () => {
             const item = wrapper.find('.item').at(8);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('Tel: 123-456-7890')
           });

            it('shows item adresse ', () => {
             const item = wrapper.find('.item').at(9);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('500 Terry Francois Street')
           });

             

});

describe('SOCIAL section', () => {

             it('it shows the social buttons ', () => {
             const item = wrapper.find('.content').at(3);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('SOCIAL')
            });

           it('shows the facebook icon ', () => {
             const item = wrapper.find('.facebook icon').first();
             expect(item.length).toEqual(1);
           });

           it('shows the twitter icon ', () => {
             const item = wrapper.find('.twitter icon').first();
             expect(item.length).toEqual(1);
           });


             it('shows the linkedin icon', () => {
             const item = wrapper.find('.linkedin icon').first();
             expect(item.length).toEqual(1);
           });

           
             it('shows the google plus icon', () => {
             const item = wrapper.find('.google plus icon').first();
             expect(item.length).toEqual(1);
           });

});

          
           const tree = renderer.create(
              <Provider store={store}>
                      <Demo />
               </Provider>
                          ).toJSON();
                  expect(tree).toMatchSnapshot();
               



});

