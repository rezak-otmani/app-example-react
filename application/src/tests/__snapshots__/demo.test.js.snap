// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="pusher"
>
  <div
    className="ui vertical masthead center aligned segment"
  >
    <div
      className="center aligned column"
    >
      <h1
        className="ui  header"
      >
        <div
          className="content"
        >
          REQUEST DEMO
          <div
            className="sub header"
          >
            <h2>
               Please fill your contact details below:
            </h2>
          </div>
        </div>
      </h1>
    </div>
    <div
      className="row"
    >
      <div
        className="ui container"
      >
        <div
          className="center aligned column"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div>
              <input
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Nom"
                type="text"
                value=""
              />
            </div>
            <br />
            <div>
              <input
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />
            </div>
            <br />
            <div>
              <input
                name="subject"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Sujet"
                type="text"
                value=""
              />
            </div>
            <br />
            <div>
              <input
                name="phone"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Téléphone"
                type="text"
                value=""
              />
            </div>
            <br />
            <div>
              <input
                name="adresse"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Adresse"
                type="text"
                value=""
              />
            </div>
            <br />
            <div>
              <button
                className="massive ui button"
                disabled={true}
                onClick={[Function]}
                type="button"
              >
                Clear Values
              </button>
              <button
                className="positive massive ui button"
                disabled={false}
                type="submit"
              >
                Send
                <i
                  className="right arrow icon"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
    <br />
    <div
      className="ui vertical olive inverted footer segment"
    >
      <div
        className="ui seven columns center aligned stackable grid"
      >
        <div
          className="three wide column"
        >
          <a
            href=""
            tabIndex="-1"
            target="_self"
          >
            <svg
              className="logo"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="-7.7116809713429575 -16.75102391736263 313.60835950127966 385.58404856714714"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                className="stylesvgbottom"
              >
                <path
                  d="M298.185 264.061l-149.093 88.021L0 264.061V88.021L149.092 0l149.093 88.021v176.04z"
                />
                <text
                  x="150"
                  y="190"
                >
                  <tspan>
                    DATO
                  </tspan>
                </text>
              </g>
            </svg>
          </a>
        </div>
        <div
          className="two wide column"
        >
          <h2
            className="ui left aligned header"
          >
            <div
              className="ui divider"
            />
            <div
              className="content"
            >
              LINKS
              <div
                className="sub header"
              >
                <div
                  className="ui link list"
                >
                  <br />
                  <a
                    className="item active"
                    href="#"
                    onClick={[Function]}
                  >
                    Home 
                  </a>
                  <a
                    className="item"
                    href="#product"
                    onClick={[Function]}
                  >
                    Product
                  </a>
                  <a
                    className="item"
                    href="#about"
                    onClick={[Function]}
                  >
                    About
                  </a>
                  <a
                    className="item"
                    href="#featured"
                    onClick={[Function]}
                  >
                    Featured
                  </a>
                  <a
                    className="item"
                    href="#contact"
                    onClick={[Function]}
                  >
                    Contact
                  </a>
                  <a
                    className="item"
                    href="#demo"
                    onClick={[Function]}
                  >
                    Demo
                  </a>
                </div>
              </div>
            </div>
          </h2>
        </div>
        <div
          className="two wide column"
        >
          <h2
            className="ui left aligned header"
          >
            <div
              className="ui divider"
            />
            <div
              className="content"
            >
              ABOUT
              <div
                className="sub header"
              >
                <div
                  className="ui link list"
                >
                  <br />
                  <a
                    className="item"
                    href=""
                  >
                    info@mysite.com 
                  </a>
                  <a
                    className="item"
                    href=""
                  >
                    info@mysite.com
                  </a>
                  <p
                    className="item"
                  >
                    Tel: 123-456-7890
                  </p>
                  <p
                    className="item"
                  >
                    500 Terry Francois Street
                  </p>
                  <p
                    className="item"
                  >
                    San Francisco, CA 94158
                  </p>
                </div>
              </div>
            </div>
          </h2>
        </div>
        <div
          className="two wide column"
        >
          <h2
            className="ui left aligned header"
          >
            <div
              className="ui divider"
            />
            <div
              className="content"
            >
              SOCIAL
              <div
                className="sub header"
              >
                <br />
                <button
                  className="ui circular facebook icon button"
                >
                  <i
                    className="facebook icon"
                  />
                </button>
                <button
                  className="ui circular twitter icon button"
                >
                  <i
                    className="twitter icon"
                  />
                </button>
                <button
                  className="ui circular linkedin icon button"
                >
                  <i
                    className="linkedin icon"
                  />
                </button>
                <button
                  className="ui circular google plus icon button"
                >
                  <i
                    className="google plus icon"
                  />
                </button>
              </div>
            </div>
          </h2>
        </div>
      </div>
    </div>
    <div
      className="ui vertical wide left aligned segment"
    >
      © 2018 by Rezak. App created with ReactJS
    </div>
  </div>
</div>
`;
